addons/base_action_rule/models/base_action_rule.py

```diff
@@ -0,0 +1,396 @@
+
+    # which fields have an impact on the registry
+    CRITICAL_FIELDS = ['model_id', 'active', 'kind', 'on_change_fields']
+
+    @api.onchange('model_id')
+    def onchange_model_id(self):
+        self.filter_pre_id = self.filter_id = False
+
+    @api.onchange('kind')
+    def onchange_kind(self):
+        if self.kind in ['on_create', 'on_create_or_write', 'on_unlink']:
+            self.filter_pre_id = self.filter_pre_domain = self.trg_date_id = self.trg_date_range = self.trg_date_range_type = False
+        elif self.kind in ['on_write', 'on_create_or_write']:
+            self.trg_date_id = self.trg_date_range = self.trg_date_range_type = False
+        elif self.kind == 'on_time':
+            self.filter_pre_id = self.filter_pre_domain = False
+
+    @api.onchange('filter_pre_id')
+    def onchange_filter_pre_id(self):
+        self.filter_pre_domain = self.filter_pre_id.domain
+
+    @api.onchange('filter_id')
+    def onchange_filter_id(self):
+        self.filter_domain = self.filter_id.domain
+
+    @api.model
+    def create(self, vals):
+        base_action_rule = super(BaseActionRule, self).create(vals)
+        self._update_cron()
+        self._update_registry()
+        return base_action_rule
+
+    @api.multi
+    def write(self, vals):
+        res = super(BaseActionRule, self).write(vals)
+        if set(vals).intersection(self.CRITICAL_FIELDS):
+            self._update_cron()
+            self._update_registry()
+        return res
+
+    @api.multi
+    def unlink(self):
+        res = super(BaseActionRule, self).unlink()
+        self._update_cron()
+        self._update_registry()
+        return res
+
+    def _update_cron(self):
+        """ Activate the cron job depending on whether there exists action rules
+            based on time conditions.
+        """
+        cron = self.env.ref('base_action_rule.ir_cron_crm_action', raise_if_not_found=False)
+        return cron and cron.toggle(model=self._name, domain=[('kind', '=', 'on_time')])
+
+    def _update_registry(self):
+        """ Update the registry after a modification on action rules. """
+        if self.env.registry.ready:
+            # for the sake of simplicity, simply force the registry to reload
+            self.env.cr.commit()
+            self.env.reset()
+            RegistryManager.new(self.env.cr.dbname)
+            RegistryManager.signal_registry_change(self.env.cr.dbname)
+
+    def _get_actions(self, records, kinds):
+        """ Return the actions of the given kinds for records' model. The
+            returned actions' context contain an object to manage processing.
+        """
+        ctx = dict(self.env.context)
+        if '__action_done' not in self.env.context:
+            ctx.update(__action_done={})
+        return self.with_context(ctx, active_test=True).search([('model', '=', records._name), ('kind', 'in', kinds)])
```

Comment:
BEWARE: You have removed some business logic from this method!
This method was already in the new API, please restore it to what it was!!!